Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs

```diff
@@ -366,23 +363,36 @@ public override void Enable()
             //NOTE: We update the source state data, in case an app wants to query it on source detected.
             for (var i = 0; i < numInteractionManagerStates; i++)
             {
-                var controller = GetController(interactionManagerStates[i].source);
-
-                if (controller != null)
-                {
-                    controller.UpdateController(interactionManagerStates[i]);
-                    inputSystem?.RaiseSourceDetected(controller.InputSource, controller);
-                }
+                CreateController(interactionManagerStates[i]);
             }
 
-            if ((inputSystem != null) &&
-                InputSystemProfile.GesturesProfile != null &&
+            if (InputSystemProfile.GesturesProfile != null &&
                 InputSystemProfile.GesturesProfile.WindowsGestureAutoStart == AutoStartBehavior.AutoStart)
             {
                 GestureRecognizerEnabled = true;
             }
         }
 
+        private void CreateController(InteractionSourceState interactionSourceState)
+        {
+            // If this is a new detected controller, raise source detected event with input system
+            // check needs to be here because GetController adds it to the activeControllers Dictionary
+            // this could be cleaned up because that's not clear
```

Comment:
Thanks for the comment. This helps a lot. I didn't realize GetController adds to activeCOntrollers

Reply:
Ah yeah, in some classes it's named `GetOrAddController`. Some inconsistent naming we can clean up.